simulation/model.py

```diff
@@ -68,13 +71,20 @@ def heating_systems(self) -> Set[HeatingSystem]:
     @property
     def air_source_heat_pump_discount_factor(self) -> float:
 
-        if self.current_datetime.year < 2022:
-            return 1
-        if self.current_datetime.year > 2022:
-            return 1 - self.air_source_heat_pump_discount_factor_2022
-        else:
-            month = self.current_datetime.month
-            return 1 - (month / 12 * self.air_source_heat_pump_discount_factor_2022)
+        if self.air_source_heat_pump_price_discount_schedule:
+
+            step_dates, discount_factors = [
+                step for step in zip(*self.air_source_heat_pump_price_discount_schedule)
+            ]
```

Comment:
This might be too long to fit on one line, and happy to leave as is. Or, you could use a walrus in the above if statement e.g `if schedule := self.air_source_heat_pump_price_discount_schedule` and then refer to it as `schedule` below (but then our code will only be `PYTHON_VERSION>=3.8`
```suggestion
            step_dates, discount_factors = zip(*self.air_source_heat_pump_price_discount_schedule)
```